pyanaconda/modules/payload/live/live_image_interface.py

```diff
@@ -76,3 +103,70 @@ def VerifySSL(self) -> Bool:
     def SetVerifySSL(self, verify_ssl: Bool):
         """Set if the ssl verification should be enabled."""
         self.implementation.set_verifyssl(verify_ssl)
+
+    @property
+    def RequiredSpace(self) -> UInt64:
+        """Space required by the source image."""
+        return self.implementation.required_space
+
+    @emits_properties_changed
+    def SetRequiredSpace(self, required_space: UInt64):
+        """Set the space required by the source image."""
+        self.implementation.set_required_space(required_space)
+
+    def UpdateKernelVersionList(self):
+        """Update the list of kernel versions."""
+        self.implementation.update_kernel_version_list()
+
+    def GetKernelVersionList(self) -> List[Str]:
+        """Get the kernel versions list."""
+        return self.implementation.kernel_version_list
+
+    @dbus_signal
+    def KernelVersionListChanged(self, kernel_version_list: List[Str]):
+        """Signal kernel version list change."""
+        pass
+
+    def SetupWithTask(self) -> ObjPath:
+        """Setup the handler.
+
+        FIXME: use real name? CheckSourceImageWithTask
+        Check availability of the image and update required space
+        """
+        return self.implementation.check_source_image_with_task()
+
+    def PreInstallWithTask(self) -> ObjPath:
```

Comment:
The methods for the installation tasks should be eventually provided by the main module.

Reply:
I'm sure about that but I guess we will see that in the future when there will be more modules implemented.

Reply:
Yeah, I didn't aspire do it right at this point.